nansat/mappers/mapper_sentinel1_l1.py

```diff
@@ -496,6 +496,8 @@ def _init_empty(self, manifest_data, annotation_data):
         self.dataset.SetMetadataItem('data_center', json.dumps(pti.get_gcmd_provider('ESA/EO')))
         self.dataset.SetMetadataItem('iso_topic_category', json.dumps(pti.get_iso19115_topic_category('Oceans')))
         self.dataset.SetMetadataItem('summary', platform_name + ' SAR data')
+
+        self.dataset.SetMetadataItem('entry_id', filename.split(".", -1)[0].split("/",-1)[-1])
```

Comment:
There is a dedicated functions in the `os.path` module to get a file's base name:
```
os.path.basename(filename).split('.')[0]
```
(There is also one to strip extensions but it does not work if there are more than one dot in the file name)

Reply:
that -1 as the second argument of ```.split(".", -1)[0]``` executes the splitting for the several times. This is not restricted unlike the dedicated function that you mentioned. Kindly, with this explanation, it is better to keep it with general splitting or with dedicated functions?

Reply:
Here, it is probably safer to keep using the `split()` method in case we get a file with several dots in its name.

FYI, you don't have to specify `-1` as the second argument of `split()`, it is the default behavior.

Reply:
Always use dedicated functions. Don't reinvent the bike:
```suggestion
        self.dataset.SetMetadataItem('entry_id',  os.path.splitext(os.path.basename(filename))[0])
```